File: trashguy.py

from time import sleep
from sys import argv


class TrashGuy:
    def __init__(self, user_input, speed=1):
        self.user_input = user_input
        self.speed = 0.1 / speed
        self.trash = "🗑"
        self.right = "(> ^_^)>"
        self.left = "<(^_^ <)"
        self.l_space = " "
        self.r_space = " "
        self.user = self.right

    def print(self):
        print(f"{self.trash}{self.l_space}{self.user}{self.r_space}{self.user_input}", end="\r")
        sleep(self.speed)

    def go_right(self):
        self.print()
        while self.r_space:
            self.r_space = self.r_space[1:]
            self.l_space += " "
            self.print()

    def go_left(self):
        self.print()
        while self.l_space:
            self.l_space = self.l_space[1:]
            self.r_space += " "
            self.print()

    def take_letter(self):
        self.letter = self.user_input[0]
        self.user = self.letter + self.left
        self.user_input = self.user_input[1:]
        self.r_space += " "
        self.l_space = self.l_space[1:]

    def trash_letter(self):
        self.user = self.user[1:]
        self.r_space += " "
        self.print()
        self.user = self.right
        self.print()
        self.l_space += " "
        self.r_space = self.r_space[1:]

    def play(self):
        for _ in self.user_input:
            self.go_right()
            self.take_letter()
            self.go_left()
            self.trash_letter()


if __name__ == "__main__":
    if len(argv) == 2:
        TrashGuy(argv[1]).play()
    elif len(argv) == 3:
        if argv[2].isnumeric():
            TrashGuy(argv[1], int(argv[2])).play()
        else:
            print("Second argument must be speed (Default=1)")
    else:
        print(f"Syntax:\npython {argv[0]} [WORD]<str> <[SPEED]<int>")

File: trashguy.cpp
#include <iostream>
#ifdef _WIN32
    #include <windows.h>

    void sleep(float milliseconds) {
        Sleep(milliseconds);
    }
#else
    #include <unistd.h>

    void sleep(float milliseconds) {
        usleep(milliseconds * 1000); // takes microseconds
    }
#endif

class TrashGuy {

    public:

        std::string right = "(> ^_^)>";
        std::string left = "<(^_^ <)";
        std::string l_space = " ";
        std::string r_space = " ";
        std::string trash = "🗑";
        std::string user = right;
        std::string user_input;
        std::string letter;
        float speed = 100;

        TrashGuy(std::string ui, float s=1) {
            user_input = ui;
            speed /= s;
        }

        void print() {
            std::cout << trash << l_space << user << r_space << user_input << "\r" << std::flush;
            sleep(speed);
        }

        void goRight() {
            print();
            while (!r_space.empty()) {
                r_space = r_space.substr(1, r_space.size()-1);
                l_space += " ";
                print();
            }
        }

        void goLeft() {
            print();
            while (!l_space.empty()) {
                l_space = l_space.substr(1, l_space.size()-1);
                r_space += " ";
                print();
            }
        }

        void takeLetter() {
            letter = user_input.at(0);
            user = letter + left;
            user_input = user_input.substr(1, user_input.size()-1);
            l_space = l_space.substr(1, l_space.size()-1);
            r_space += " ";
        }

        void trashLetter() {
            user = user.substr(1, user.size()-1);
            r_space += " ";
            print();
            user = right;
            print();
            l_space += " ";
            r_space = r_space.substr(1, r_space.size()-1);
        }

        void play() {
            for (char _ : user_input) {
                goRight();
                takeLetter();
                goLeft();
                trashLetter();
            }
        }
};

int main(int argc, char *argv[]) {
    if (argc == 2) {
        TrashGuy(argv[1]).play();
    } else if (argc == 3) {
        TrashGuy(argv[1], std::stof(argv[2])).play();
    }
}
